_studio/mfx_lib/encode_hw/h265/include/mfx_h265_encode_vaapi.h

```diff
@@ -181,6 +181,30 @@ mfxStatus SetSkipFrame(
 
             void Init(mfxU32 picWidthInLumaSamples, mfxU32 picHeightInLumaSamples);
     };
+
+    class GUIDhash
+    {
+    public:
+        size_t operator()(const GUID &guid) const
+        {
+            return guid.GetHashCode();
+        }
+    };
+
+    struct VAParameters
+    {
+        VAParameters():
+            profile(VAProfileNone), entrypoint(static_cast<VAEntrypoint>(0))
+        {}
+
+        VAParameters(VAProfile p, VAEntrypoint e) :
+            profile(p), entrypoint(e)
+        {}
+
+        VAProfile profile;
```

Comment:
defaults can be written here to save some code lines occupied by default constructor :)